scanner/src/main/kotlin/Scanner.kt

```diff
@@ -235,6 +238,8 @@ class Scanner(
                     async {
                         pkg to runCatching {
                             packageProvenanceResolver.resolveProvenance(pkg, downloaderConfig.sourceCodeOrigins)
+                        }.onFailure {
+                            if (it is CancellationException) currentCoroutineContext().ensureActive()
```

Comment:
Can you explain why this is necessary here, i.e., what is the problem with ignoring the exception?

Reply:
I guess you wrote this before the above summary post, so I'll only answer to the above.

Reply:
I was especially curious about the case of ignoring cancellation exceptions like we did here before. IIUC this can cause the current coroutine context to not get cancelled properly, right?